src/cloudwatch.ts

```diff
@@ -1,79 +1,152 @@
+      .describeLogGroups({
+        nextToken,
+        logGroupNamePrefix,
+      })
+      .promise();
+    const newAcc = acc.concat(result.logGroups ?? []);
+    if (result.nextToken) {
+      return rec(newAcc, result.nextToken);
+    } else {
+      return newAcc;
     }
-    return rec([]);
+  }
+  return rec([]);
 }
 
-export async function subscribeGroups(cloudwatchLogs: CloudWatchLogs, groups: CloudWatchLogs.LogGroup[], filterName: string, targetLambda: string): Promise<void> {
-    await Promise.all(groups.map(async (group) => {
-        const subscriptions = await getSubscriptions(cloudwatchLogs, group.logGroupName!, filterName);
-        const subscriptionExists = subscriptions.some(sub => sub.filterName === filterName && sub.destinationArn === targetLambda);
-        if (!subscriptionExists) {
-            console.log(`Adding subscription for ${group.logGroupName!} with name ${filterName}`);
-            await putSubscription(cloudwatchLogs, group.logGroupName!, filterName, targetLambda);
-        }
-    }));
+export async function subscribeGroups(
+  cloudwatchLogs: CloudWatchLogs,
+  groups: CloudWatchLogs.LogGroup[],
+  filterName: string,
+  targetLambda: string
+): Promise<void> {
+  await Promise.all(
+    groups.map(async (group) => {
+      const subscriptions = await getSubscriptions(
+        cloudwatchLogs,
+        group.logGroupName!,
+        filterName
+      );
+      const subscriptionExists = subscriptions.some(
+        (sub) =>
+          sub.filterName === filterName && sub.destinationArn === targetLambda
+      );
+      if (!subscriptionExists) {
+        console.log(
+          `Adding subscription for ${group.logGroupName!} with name ${filterName}`
+        );
+        await putSubscription(
+          cloudwatchLogs,
+          group.logGroupName!,
+          filterName,
+          targetLambda
+        );
+      }
+    })
+  );
 }
 
-export async function unsubscribeGroups(cloudwatchLogs: CloudWatchLogs, groups: CloudWatchLogs.LogGroup[], filterName: string): Promise<void> {
-    await Promise.all(groups.map(async (group) => {
-        const subscriptions = await getSubscriptions(cloudwatchLogs, group.logGroupName!, filterName);
-        subscriptions.forEach ( async (subscription) => {
-            if (subscription.filterName === filterName) {
-                console.log(`Removing subscription for ${group.logGroupName!} with name ${filterName}`);
-                await deleteSubscription(cloudwatchLogs, group.logGroupName!, filterName);
-            }
-        });
-    }));
+export async function unsubscribeGroups(
+  cloudwatchLogs: CloudWatchLogs,
+  groups: CloudWatchLogs.LogGroup[],
+  filterName: string
+): Promise<void> {
+  await Promise.all(
+    groups.map(async (group) => {
+      const subscriptions = await getSubscriptions(
+        cloudwatchLogs,
+        group.logGroupName!,
+        filterName
+      );
+      for (const subscription of subscriptions) {
```

Comment:
This was changed from ` subscriptions.forEach ( async (subscription) => {`, as it was causing problems with `Promise returned in function argument where a void return was expected.` error. I _think_ this change is fine, but this is the only change made manually that I'm not 100% sure on.